### ISSUE TITLE
Focus not working in FunSpecs

### ISSUE THREAD

**Issue by ps-feng** [kotest/kotest#1376]
This is not working for FunSpecs:

```kotlin
internal class SomeTest : FunSpec() {
    init {
        context("f:my test") {
            test("something") { ... }
        }
    }
}
```
According to the [documentation](https://github.com/kotest/kotest/blob/master/doc/conditional_evaluation.md#focus), wouldn't `context` be considered a top-level scope and thus should work? Or is this a special case?

**Comment by sksamuel** [comment]
What is the expected behavior in this case. You're focusing the only test, so it should run?

**Comment by ps-feng** [comment]
The example was not explicit enough perhaps so let me try again. Ideally I'd be able to focus on a single test but I was under the impression this is supported:
```kotlin
internal class SomeTest : FunSpec() {
    init {
        context("f:my test 1") {
            test("something 1") { ... }
            test("something 2") { ... }
        }
        context("my test 2") {
            test("blabla") { ... }
            test("lala") { ... }
        }
    }
}
```
I'd expect `something 1` and `something 2` to be executed, as the top-level `f:my test 1` is focused.

**Comment by sksamuel** [comment]
Yes that should work. I'll add your test and see.

**Comment by sksamuel** [comment]
I've added test and they pass. Are you on 4.0.2 ?

**Comment by ps-feng** [comment]
I am on 4.0.2. I've just tested this and it doesn't work:
```kotlin
internal class BlaBlaTest : FunSpec() {

    init {
        context("f:some test") {
            test("should run") {
                1 shouldBe 1
            }
            test("should run 2") {
                2 shouldBe 1
            }
        }
        context("other test") {
            test("should never run") {
                3 shouldBe 4
            }
        }
    }
}
```
![image](https://user-images.githubusercontent.com/13632227/79054698-5a6f7e80-7c47-11ea-8765-707d91cf4175.png)
![image](https://user-images.githubusercontent.com/13632227/79054563-4414f300-7c46-11ea-88f5-8e3c9903b59c.png)

**Comment by sksamuel** [comment]
Thanks for this bug report. It is now fixed in master and will be included in 4.0.5.
